docs/topics/secrets.md

```diff
@@ -115,13 +115,135 @@ adopted into Kubernetes, we plan to support them. However, the present stable
 version of Kubernetes Secrets only Base64-encodes data.
 
 Our present recommendation is for Brigade developers to fetch the secret directly from a
-trusted key store such as Vault.
+trusted key store such as Vault.  See the [example](#fetching-secrets-from-a-remote-store) below.
 
 Alternatively, you could use `secretKeyRef` to reference existing secrets already in your
-Kubernetes cluster.
+Kubernetes cluster.  See the [example](#using-kubernetes-secrets) below.
 
 **I don't want to use Helm to manage my project/secrets. Can I do it manually?**
 
 Yes. Helm is there to make your life easier, but you can manage project secrets manually.
 You cannot, however, make manual modifications and then go back to using Helm. Doing so
 may result in lost data.
+
+## Examples
+
+---
+
+## Fetching secrets from a remote store
+
+Here is an example where we fetch secrets from a remote store
+([Azure Key Vault](https://azure.microsoft.com/en-us/services/key-vault/))
+for use by one or more Jobs.
+
+To do so, we first run a Job to authenticate with the remote store, fetch all needed secrets
+and save these secret values to a shared directory.  Then, we run a Job that consumes these values.
+
+Here's what our `brigade.js` file looks like:
+
+```javascript
+const { events, Job, Group } = require("brigadier");
+
+const sharedMountPrefix = `/mnt/brigade/share`;
+const secrets = [
+  "foo",
+  "bar"
+]
+
+events.on("exec", (event, project) => {
+  // Create Job to fetch secrets
+  secretfetcher = new Job("secretfetcher", "microsoft/azure-cli:latest");
+  secretfetcher.storage.enabled = true;
+
+  // Login as an Azure Service Principal
+  secretfetcher.tasks.push(
+    `az login --service-principal \
+    -u ${project.secrets.spID} \
+    -p '${project.secrets.spPW}' \
+    --tenant ${project.secrets.spTenant}`)
+
+  // Fetch all secrets from the Azure Key Vault
+  for (i in secrets) {
+    secretfetcher.tasks.push(
+      `az keyvault secret show --vault-name ${project.secrets.keyvault} \
+      -n ${secrets[i]} --query value > ${sharedMountPrefix}/${secrets[i]}`);
+  }
+
+  // Create Job to consume secrets
+  secretconsumer = new Job("secretconsumer", "alpine");
+  secretconsumer.storage.enabled = true;
+
+  // Consume all secrets
+  for (i in secrets) {
+    secretconsumer.tasks.push(
+      `ls -haltr ${sharedMountPrefix}/${secrets[i]}`);
+  }
+
+  // Run jobs sequentially
+  Group.runEach([secretfetcher, secretconsumer]);
+})
+```
+
+To run this example, we replace `BRIGADE_PROJECT_NAME` with our Brigade project name and run:
+
+```console
+brig run BRIGADE_PROJECT_NAME -f brigade.js
+```
+
+When we check the logs of the `secretconsumer` job pod, we see:
+
+```console
+ $ k logs secretconsumer-01d23ch2n6d1847ys6x48x1rez
+-rwxrwxrwx    1 root     root           6 Jan 25 20:52 /mnt/brigade/share/foo
+-rwxrwxrwx    1 root     root           6 Jan 25 20:52 /mnt/brigade/share/bar
+```
+
+## Using Kubernetes secrets
```

Comment:
Credit to @dgkanatsios for this example :)

Reply:
Maybe ‘kubectl’ instead of ‘k’? (I do have the same alias btw!)

Reply:
oops!  good catch; updated.